diff --git a/argp-standalone/argp-fmtstream.h b/argp-standalone/argp-fmtstream.h
index 6198cde..f3cee24 100644
--- a/argp-standalone/argp-fmtstream.h
+++ b/argp-standalone/argp-fmtstream.h
@@ -28,7 +28,13 @@
 
 #include <stdio.h>
 #include <string.h>
+#if HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+
+#ifdef _WIN32
+typedef ptrdiff_t ssize_t;
+#endif
 
 #if _LIBC || (defined (HAVE_FLOCKFILE) && defined(HAVE_PUTC_UNLOCKED) \
      && defined (HAVE_FPUTS_UNLOCKED) && defined (HAVE_FWRITE_UNLOCKED) )
diff --git a/argp-standalone/argp-parse.c b/argp-standalone/argp-parse.c
index 10bff92..25d44a2 100644
--- a/argp-standalone/argp-parse.c
+++ b/argp-standalone/argp-parse.c
@@ -43,7 +43,9 @@ char *alloca ();
 
 #include <stdlib.h>
 #include <string.h>
+#if HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <limits.h>
 #include <assert.h>
 
